### PR TITLE
Build Sway projects at every CI run

### PR DESCRIPTION
**Pull request by digorithm**
This PR: 

1. Removes all lock files, ABI files, and binary files for test projects used in the `harness.rs` test suite;
2. Adds respective `gitignore`s to not include lock files, ABI files, and binary files;
3. Adds a Rust binary to build all test projects (`cargo run --bin build-test-projects`). This needs to be run locally when running tests;
4. Updates the CI workflow file to run the above binary in order to build all test projects, from scratch, using the latest `forc`, at every new CI run, guaranteeing fresh binaries when running the test suite.

Closes https://github.com/FuelLabs/fuels-rs/issues/210.
Closes https://github.com/FuelLabs/fuels-rs/issues/201.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (vnepveu) [inline .github/workflows/ci.yml:100]: we would need the new `fuels-helpers` crate here now, no?

File: packages/fuels-abigen-macro/tests/test_projects/contract_output_test/Forc.lock
Comment (vnepveu) [inline packages/fuels-abigen-macro/tests/test_projects/contract_output_test/Forc.lock:2]: This file should be removed? unless it's special

File: packages/fuels-abigen-macro/tests/test_projects/swayswap/swayswap_abi/Forc.toml
Comment (vnepveu) [inline packages/fuels-abigen-macro/tests/test_projects/swayswap/swayswap_abi/Forc.toml:2]: I don't think the `swayswap` project should be in there

File: packages/fuels-abigen-macro/tests/test_projects/swayswap/swayswap_abi/src/main.sw
Comment (vnepveu) [inline packages/fuels-abigen-macro/tests/test_projects/swayswap/swayswap_abi/src/main.sw:2]: same

File: scripts/build-test-projects/Cargo.toml
Comment (vnepveu) [inline scripts/build-test-projects/Cargo.toml:6]: add newline

**Review by vnepveu — CHANGES_REQUESTED** [review]
Just remove the swayswap project and add some newlines, otherwise great work!!! This is going to be awesome to have!

File: packages/fuels-abigen-macro/tests/test_projects/contract_output_test/Forc.lock
Comment (digorithm) [reply packages/fuels-abigen-macro/tests/test_projects/contract_output_test/Forc.lock:2]: Oh no, it should!

File: packages/fuels-abigen-macro/tests/test_projects/swayswap/swayswap_abi/Forc.toml
Comment (digorithm) [reply packages/fuels-abigen-macro/tests/test_projects/swayswap/swayswap_abi/Forc.toml:2]: I believe it should. It's a massive test case for the SDK, it covers a _lot_ of use cases.

**Comment by Voxelot** [comment]
You could consider using the somewhat idiomatic `xtask` approach for these build scripts.

https://github.com/matklad/cargo-xtask

We use it in fuel-core and it works great.

**Comment by digorithm** [comment]
> You could consider using the somewhat idiomatic `xtask` approach for these build scripts.
> 
> https://github.com/matklad/cargo-xtask
> 
> We use it in fuel-core and it works great.

That'd be a nice idea to explore in future PRs -- I have no idea how it works and it would require some exploration time on my end, plus: 
> cargo-xtask is neither an officially recommended workflow, nor a de-facto standard (yet?). It might or might not work for your use case.

For now, I'd rather push this out and unblock _many_ things that are waiting for the next release.

**Review by vnepveu — DISMISSED** [review]
LGTM :tada: :tada:

File: .github/workflows/ci.yml
Comment (adlerjohn) [inline .github/workflows/ci.yml:48]: You should fix a `forc` version here instead of using latest

**Review by adlerjohn — CHANGES_REQUESTED** [review]
above

**Review by vnepveu — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (vnepveu) [reply .github/workflows/ci.yml]: Wait isn't it going to be problematic? I feel like the whole point was to use the last compiler version (ie last `forc` version) so we shouldn't pin it?

Comment (digorithm) [reply .github/workflows/ci.yml]: Yeah, ideally we want to test it against the latest. But, for instance, if for some reason we don't wanna use the latest (e.g. latest breaks something on the SDK, we're aware of it, but we're still working on a fix), we might not want to block other PR's CI pipeline because of it. So having this control is nice. 

But, then again, we have to watch ourselves and make sure we're updating this version when there's a new `forc` version.

Comment (adlerjohn) [reply .github/workflows/ci.yml]: If we want the best of both worlds, then we can create a second CI workflow that runs against `latest` `forc` and isn't required to pass to merge PRs. Then if it breaks we know, but it's not blocking anything.

Comment (vnepveu) [reply .github/workflows/ci.yml]: Ok, I feel like we should find a way to ensure this?

Comment (digorithm) [reply .github/workflows/ci.yml]: Not a huge fan of that because I believe it would increase the CI runtime considerably :(. We're currently sitting at ~10min, running another one, in this case, would mean building all Sway projects again.

Comment (vnepveu) [reply .github/workflows/ci.yml]: maybe the solution would be to run the `fuels-rs` CI on every `forc` major release?
But just on releases?

Comment (vnepveu) [reply .github/workflows/ci.yml]: Ah no wait that doesn't work

**Pull request merged**
